taichi/backends/metal/shaders/ad_stack.metal.h

```diff
@@ -0,0 +1,76 @@
+#include "taichi/backends/metal/shaders/prolog.h"
+
+#ifdef TI_INSIDE_METAL_CODEGEN
+
+#ifndef TI_METAL_NESTED_INCLUDE
+#define METAL_BEGIN_RUNTIME_STRUCTS_DEF constexpr auto kMetalAdStackSourceCode =
```

Comment:
nit: Shall we use the `TI_` prefix?

Reply:
TLDR is no... This is a somewhat obscure mechanism to support both including these this code as C++ code, and stringifying them as real Metal kernel/shader code. Every code in `taichi/backends/metal/shaders` is following this style. The macros here are only meant to be used by the Metal backend, and not shared to a broader scope.